plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/JdbcIcebergClient.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.jdbc;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.connector.SchemaTableName;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+
+import java.util.AbstractMap.SimpleEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+public class JdbcIcebergClient
+{
+    private final JdbcIcebergConnectionFactory connectionFactory;
+    private final String catalogId;
+
+    public JdbcIcebergClient(JdbcIcebergConnectionFactory connectionFactory, String catalogId)
+    {
+        this.connectionFactory = requireNonNull(connectionFactory, "connectionFactory is null");
+        this.catalogId = requireNonNull(catalogId, "catalogId is null");
+    }
+
+    public List<String> getNamespaces()
+    {
+        ImmutableList.Builder<String> namespaces = ImmutableList.builder();
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            namespaces.addAll(handle.createQuery("" +
```

Comment:
`JdbcCatalog.listNamespaces` wouldn't work if my understanding is correct. We want to list all namespaces without providing the name, but the method expects the argument. If we call `listNamespaces()` instead of `listNamespaces(Namespace namespace)`, it fails with schema existence check.

Reply:
@ebyhr That PR has been merged. Isn't this let us use `listNamespaces()`?